.buildkite/pipeline.yml

```diff
@@ -1,77 +1,144 @@
 x-common-params:
-  &publish-android-artifacts-docker-container
-  docker#v3.8.0:
-    image: "public.ecr.aws/automattic/android-build-image:v1.1.0"
-    propagate-environment: true
-    environment:
-      # DO NOT MANUALLY SET THESE VALUES!
-      # They are passed from the Buildkite agent to the Docker container
-      - "AWS_ACCESS_KEY"
-      - "AWS_SECRET_KEY"
+  - &gb-mobile-docker-container
+    docker#v3.8.0:
+      image: "public.ecr.aws/automattic/gb-mobile-image:latest"
+      environment:
+        - "CI=true"
+        # Allow WP-CLI to be run as root, otherwise it throws an exception.
+        # Reference: https://git.io/J9q2S
+        - "WP_CLI_ALLOW_ROOT=true"
+  - &publish-android-artifacts-docker-container
+    docker#v3.8.0:
+      image: "public.ecr.aws/automattic/android-build-image:v1.1.0"
+      propagate-environment: true
+      environment:
+        # DO NOT MANUALLY SET THESE VALUES!
+        # They are passed from the Buildkite agent to the Docker container
+        - "AWS_ACCESS_KEY"
+        - "AWS_SECRET_KEY"
 
 steps:
   - block: "Request trigger Android bundle and builds"
     branches: "dependabot/submodules/*"
 
-  - label: "Build JS Bundles"
-    key: "js-bundles"
+  - label: Build npm cache
+    key: build-npm
     plugins:
-      - docker#v3.8.0:
-          image: "public.ecr.aws/automattic/gb-mobile-image:latest"
-          environment:
-            - "CI=true"
-            # Allow WP-CLI to be run as root, otherwise it throws an exception.
-            # Reference: https://git.io/J9q2S
-            - "WP_CLI_ALLOW_ROOT=true"
+      - automattic/git-shallow-clone
     command: |
-        source /root/.bashrc
-
-        echo "--- :node: Setup Node environment"
-        nvm install && nvm use
+      echo "--- :docker: Build Docker image"
+      make ci_build_docker_image
+      echo "--- :npm: Build node dependencies if needed"
+      make ci_build_dependencies
```

Comment:
First step: Either build the cache for the `npm install` output or establish one is not necessary.